features/step_definitions/password_steps.rb

```diff
@@ -56,3 +56,28 @@ def visit_url_in_email(email, subject)
   click_on "Send instructions"
   page.should have_selector(RECAPTCHA_SCRIPT, visible: false)
 end
+
+Then "it should not be possible to reset their password" do
+  assert_not has_link?('Forgot password?', visible: :all)
+
+  expect do
+    visit path_to('the provider reset password page') # FIXME: first time does not raise??
+    visit path_to('the provider reset password page')
+  end
+    .to raise_error(ActionController::RoutingError)
```

Comment:
Folow-up to the other thread

```suggestion
    assert_raises(ActionController::RoutingError) do
      visit path_to('the provider reset password page')
      Capybara.current_session.raise_server_error!
    end
```

The reason is that depending on the driver the exception is raised either immediately or only accessible through the capybara session. 

In the latter case, the exception is only raised during `AfterStep` I believe registered at `features/support/hooks.rb`
```
AfterStep do
  page.raise_server_error!
end
```

I'm not sure what makes more sense to call it on `page` or on `Capybara.current_session`. It works either way.

P.S. no idea why a second visit alters this behavior but I don't care enough to spend more time on this